src/main/java/org/testng/DataProviderInformation.java

```diff
@@ -0,0 +1,33 @@
+package org.testng;
+
+import java.lang.reflect.Method;
+
+/**
+ * Represents information pertaining to a data provider that would be invoked.
+ */
+public class DataProviderInformation {
+    private Object instance;
+    private Method method;
+
+    public DataProviderInformation(Method method, Object instance) {
```

Comment:
@cbeust Could you double check your previous comment? I think it could be a good thing to have `IDataProviderAnnotation` included and remove `DataProviderHolder` which has the exact same purpose.
And I think we should find a better name than `DataProviderInformation`.